### PR TITLE
Neodata Group support for adagio, 4w recom, 4w native ppn

### PR DESCRIPTION
**Pull request by neodatagroup**

### CODE REVIEW

File: ads/4wppn.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Please refer to [Help Desk](4winfo@4wmarketplace.com) for more
+information on how to get required ad tag or placement IDs.
+
+## Examples
+
+### Simple ad tag 
+
+```html
+ <amp-ad width=320 height=250
+          type="adagio"
```
Comment (cramforce) [inline ads/4wppn.md]: Typo?

But is there any chance you could actually do with a single ad type for all of these and switch inside the implementation based on some param?

File: ads/adagio.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  $neodata._adagio = {};
+  $neodata._adagio.Q = [];
+  $neodata._adagio.amp = data;
+
+  $neodata._adagio.Q.push(['setup', function () {
+    $neodata._adagio.B.addPosition(data.pid, 'c');
+  }]);
+
+  loadScript($neodata, 'http://ads.localhost:8000/ads/adagio_lib.js');
```
Comment (cramforce) [inline ads/adagio.js]: Please fix.

File: ads/adagio.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+
+import {loadScript, checkData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function neo4w(global, data) {
```
Comment (cramforce) [inline ads/adagio.js]: Typo?

The registration of these is missing.

**Comment by cramforce** [comment]
Please ping back when this is working locally for you.

File: ads/4wppn.md (hunk 1 above)
Comment (neodatagroup) [inline ads/4wppn.md]: Unfortunately not

File: ads/adagio.js (hunk 2 above)
Comment (neodatagroup) [inline ads/adagio.js]: Created a new pull
https://github.com/ampproject/amphtml/pull/4011

**Pull request closed without merging**